crates/core/transaction/src/plan/build.rs

```diff
@@ -413,48 +52,42 @@ impl TransactionPlan {
             memo,
         };
 
-        // Finally, compute the binding signature and assemble the transaction.
-        let binding_signing_key = rdsa::SigningKey::from(synthetic_blinding_factor);
-        let auth_hash = transaction_body.auth_hash();
-        let binding_sig = binding_signing_key.sign(rng, auth_hash.as_bytes());
-        tracing::debug!(bvk = ?rdsa::VerificationKey::from(&binding_signing_key), ?auth_hash);
-
-        Ok(UnauthTransaction {
-            inner: Transaction {
-                transaction_body,
-                binding_sig,
-                anchor: witness_data.anchor,
-            },
-            synthetic_blinding_factor,
+        Ok(Transaction {
+            transaction_body,
+            anchor: witness_data.anchor,
+            binding_sig: [0; 64].into(),
         })
     }
-}
 
-/// A partially-constructed transaction awaiting authorization data.
-pub struct UnauthTransaction {
-    inner: Transaction,
-    synthetic_blinding_factor: Fr,
-}
-
-impl UnauthTransaction {
-    pub fn authorize<R: CryptoRng + RngCore>(
-        mut self,
+    /// Slot in the [`AuthorizationData`] and derive the synthetic
+    /// blinding factors needed to compute the binding signature
+    /// and assemble the transaction.
+    pub fn apply_auth_data<R: CryptoRng + RngCore + Debug>(
+        &self,
         rng: &mut R,
         auth_data: &AuthorizationData,
+        mut transaction: Transaction,
     ) -> Result<Transaction> {
         // Do some basic input sanity-checking.
-        let spend_count = self.inner.spends().count();
+        let spend_count = transaction.spends().count();
         if auth_data.spend_auths.len() != spend_count {
             anyhow::bail!(
                 "expected {} spend auths but got {}",
                 spend_count,
                 auth_data.spend_auths.len()
             );
```

Comment:
It returns early with an error, but we can change it.

Reply:
I'm confused — bail!() does return an error rather than panicking.

Reply:
If it returns an Error type, it's ok. But I wonder if bail specifically panics/aborts. This is the behavior that isn't too safe and causes unexpected behavior given it can't be caught.

Reply:
No, bail!() specifically doesn't panic — its purpose is to have something that's ergonomically like panic!() but returns an error instead.

Reply:
Ah, just seeing Henry's comment. We are good then 👍